src/utils/SpatialTreeInterface/SpatialTreeInterface.jl

```diff
@@ -0,0 +1,226 @@
+module SpatialTreeInterface
+
+import ..LoopStateMachine: @controlflow
+
+import Extents
+import GeoInterface as GI
+import AbstractTrees
+
+# ## Interface
+# Interface definition for spatial tree types.
+# There is no abstract supertype here since it's impossible to enforce,
+# but we do have a few methods that are common to all spatial tree types.
+
+"""
+    isspatialtree(tree)::Bool
+
+Return true if the object is a spatial tree, false otherwise.
+
+## Implementation notes
+
+For type stability, if your spatial tree type is `MyTree`, you should define
+`isspatialtree(::Type{MyTree}) = true`, and `isspatialtree(::MyTree)` will forward
+to that method automatically.
+"""
+isspatialtree(::T) where T = isspatialtree(T)
+isspatialtree(::Type{<: Any}) = false
+
+
+"""
+    getchild(node)
+
+Return an iterator over all the children of a node.
+This may be materialized if necessary or available,
+but can also be lazy (like a generator).
+"""
+getchild(node) = AbstractTrees.children(node)
+
+"""
+    getchild(node, i)
+
+Return the `i`-th child of a node.
+"""
+getchild(node, i) = getchild(node)[i]
+
+"""
+    nchild(node)
+
+Return the number of children of a node.
+"""
+nchild(node) = length(getchild(node))
```

Comment:
`getchild` promises only an Iterator but these assume length and indexing

Reply:
yeah true, might have to skip this one...not sure.  maybe I should refactor get_indices_extents somehow...that's the only place that uses this at the moment I think.

It may be useful when doing dual tree branch-and-bound / NN queries though, so will have to test before removing.